neat_core.c

```diff
@@ -804,6 +1035,210 @@ neat_open(neat_ctx *mgr, neat_flow *flow, const char *name, uint16_t port)
     return neat_he_lookup(mgr, flow, he_connected_cb);
 }
 
+neat_error_code
+neat_open_localname(neat_ctx *mgr, neat_flow *flow, const char *name, uint16_t port, const char *localname)
+{
+    neat_log(NEAT_LOG_DEBUG, "%s", __func__);
+
+    if (flow->name) {
+        return NEAT_ERROR_BAD_ARGUMENT;
+    }
+
+    if (localname == NULL)
+	return neat_open(mgr, flow, name, port);
+
+    return NEAT_ERROR_UNABLE;
+}
+
+neat_error_code
+neat_open_multistream(neat_ctx *mgr, neat_flow *flow, const char *name, uint16_t port, const char* localname, int count)
```

Comment:
I wasn't thinking about variadic functions. I think that we might want to provide a flexible way for providing arguments in multiple places. So I think a consistent way within the neat API should be used.
One way I'm thinking about is to provide a pointer to a sequence of TLVs. It can be NULL, if you caller doesn't want to provide anything, it can provide multiple parameters. Pretty flexible and easy to pass. An alternative would be to use a single pointer and an enumeration type variable to indicate what is provided.
@kristrev Any preference?